compiler/lib/effects.ml

```diff
@@ -289,13 +302,30 @@ let cps_block ~st ~k pc block =
     | Some (body_prefix, last_instrs, last) ->
         List.map body_prefix ~f:(fun i -> cps_instr ~st i) @ last_instrs, last
     | None ->
-        let last_instrs, last = cps_last ~st block.branch ~k in
-        let body =
-          List.map block.body ~f:(fun i -> cps_instr ~st i)
-          @ alloc_jump_closures
-          @ last_instrs
+        let ignore_cps =
+          match block.branch with
+          | Branch (pc, _) -> (
+              match Hashtbl.find st.cfg.simple_branch pc with
+              | Simple -> true
+              | No -> false
+              | exception Not_found -> false)
+          | _ -> false
         in
-        body, last
+
+        if ignore_cps
+        then
+          let body =
+            List.map block.body ~f:(fun i -> cps_instr ~st i) @ alloc_jump_closures
+          in
+          body, block.branch
+        else
+          let last_instrs, last = cps_last ~st block.branch ~k in
+          let body =
+            List.map block.body ~f:(fun i -> cps_instr ~st i)
+            @ alloc_jump_closures
```

Comment:
I think that in this case `alloc_jump_closures` defines a single function whose body is the block we are jumping to.
```suggestion
```